### DISCUSSION TITLE
Edge functions with write access

### DISCUSSION THREAD

**Discussion by yngwi** [supabase/supabase#12102] [Questions] [answered]
Hi,
I'm wondering how to best create an edge function that will run periodically (triggered from Postres?) and is supposed to insert/update data based on external resources. Read requests to supabase from inside the function work good with the simple read access for all users RLS policy, but how do I best write a policy that let's my function write to the database? Do I create a db role 'edge_function', add insert and update policies that give users with this role access to the needed tables? How do I then get a Bearer token for this "role"? How can I use this role on both the hosted and localhost (dev) supabase instances?
Thank you!

**Comment by yngwi** [comment, marked as answer]
To answer my own question: I can use the service role key passed on to the function environment.

**Comment by deepak-coding-art** [comment]
Can we make a role and give it permission to update a single table bypassing RLS on that table and generate a Bearer token for that role to use in the edge function so the app should be more secure than using service role key it has too much power.

**Comment by vlad-ed-git** [comment]
Can we elaborate on what exactly "I can use the service role key passed on to the function environment." means.

**Reply by GaryAustin1** [reply]
The docs have examples.
https://supabase.com/docs/guides/functions/secrets discusses the env variables.
You initialize a supabase client with service_role from the environment variables if you want to access the REST API.
